Core/MIPS/x86/X64IRCompALU.cpp

```diff
@@ -185,10 +179,43 @@ void X64JitBackend::CompIR_Compare(IRInst inst) {
 
 	switch (inst.op) {
 	case IROp::Slt:
+		regs_.Map(inst);
+		CMP(32, regs_.R(inst.src1), regs_.R(inst.src2));
+		SETcc(CC_L, R(SCRATCH1));
+		MOVZX(32, 8, regs_.RX(inst.dest), R(SCRATCH1));
+		break;
+
 	case IROp::SltConst:
+		if (inst.constant == 0) {
+			// Basically, getting the sign bit.  Let's shift instead.
```

Comment:
Hm, could the other case do SUB and then shift, instead of setcc? Feels like it should work and be one instruction smaller?

Reply:
And come to think of it, that could even be done earlier in the IR pipeline (right at mips->ir time), since flags aren't used. Though, not sure if there'd be much benefit.

Reply:
I think a SUB would only work if the register was "normalized" in RISC-V terms, which won't be the case often enough.

For example, suppose it was already INT32_MIN.  If we subtract 1, it becomes INT32_MAX and the sign bit is not set - but it was lower than 1.  It would work if sign extending to 64 bits first, but that probably costs more...

-[Unknown]